packages/ra-core/src/controller/field/useReference.ts

```diff
@@ -0,0 +1,93 @@
+import { useEffect } from 'react';
+// @ts-ignore
+import { useDispatch, useSelector } from 'react-redux';
+import get from 'lodash/get';
+
+import { crudGetManyAccumulate } from '../../actions';
+import { linkToRecord } from '../../util';
+import { Record, ReduxState } from '../../types';
+
+interface Props {
```

Comment:
you shouldn't call it `props`, it's not a component

Reply:
I think I named mine `Props` too in the `useListParams`

Reply:
Btw, although I can understand this may be misleading with the React props, they are indeed props|properties of an object passed as a parameter to the hooks.